src/lug/System/Win32/PathImplWin32.cpp

```diff
@@ -0,0 +1,76 @@
+#include <lug/System/Win32/PathImplWin32.hpp>
+
+#include <Shlobj.h>
+#include <windows.h>
+
+std::string lug::System::Path::priv::getRoot() {
+    char* driveLetter = getEnv("HOMEDRIVE");
+    if (driveLetter == nullptr) {
+        return std::string("");
+    }
+
+    std::string rootPath;
+    rootPath += driveLetter;
+
+    free(driveLetter);
+
+    return rootPath;
+}
+
+std::string lug::System::Path::priv::getHome() {
+    char* driveLetter = getEnv("HOMEDRIVE");
+    char* homeDirectory = getEnv("HOMEPATH");
+
+    if (driveLetter == nullptr || homeDirectory == nullptr) {
+        return std::string("");
+    }
+
+    std::string homePath;
+    homePath += driveLetter;
+    homePath += homeDirectory;
+
+    free(driveLetter);
+    free(homeDirectory);
+
+    return homePath;
+}
+
+std::string lug::System::Path::priv::getCwd() {
+    TCHAR cwd[MAX_PATH_SIZE];
+    if (FAILED(GetCurrentDirectory(MAX_PATH_SIZE, cwd))) {
+        return std::string("");
+    }
+
+    std::basic_string<TCHAR> cwdString(cwd);
+    std::string cwdPath(cwdString);

```

Comment:
Why not `return cwd;`? Because `TCHAR` is just a `typedef` for `char`.
And you can return this local array because it will be used to construct a string.